### PR TITLE
Fixes #10005 This time with a Vengeance

### LINKED ISSUE #10005: Project Mind can be used to metagame someone's real identity
If someone masks their identity by any means, such as using someone else's ID and a mask, an agent ID, whatever have you, things get tricky and meta-ey.
Their FAKE name is displayed in the list of people you can project mind to (i.e. "Don Newman"). Up until you send the message, they are always displayed using their FAKE name.
dreamseeker_2016-05-10_16-44-47

However, the chatbox message you get afterwards uses their REAL name.
dreamseeker_2016-05-10_16-45-34

Someone using a voice changer to impersonate someone else? Let me whip out my project mind real quick so I can instantly confirm there are 2 identical names on the list. Sure there are, let me call them a faggot real quick. Oh that's who they really are.

### PR DESCRIPTION
**Pull request by ghost**
Because I couldn't sleep without amending #10677 

04:45 - 🐔🐥🐣Komrade D3athrow🐣🐥🐔: here's an idea make it so you can only remote see or remote talk to somebody you've seen once

went with heard rather than seen, though.
- bugfix: Changed behaviour of grey telepathy to not be awful and highly exploitable. 
- tweak: You now have to hear someone speak locally - in other words, you have to see their speech bubble, before you can telepathically communicate with them.
- tweak: The name chosen to represent them in the menu is the one you heard them speak as.
- tweak: You can no longer discover if someone is dead by trying to send them a message.

Fixes #10005 
**Pretty robustly tested.**

### CODE REVIEW

**Comment by Shadowmech88** [comment]
I'd personally prefer having to see them rather than hear them, but I still like the concept of this a lot more than having to type their name.

**Comment by NigglyWiggly** [comment]
What does this consider having been 'heard'?

Does it include: Radio, Cult communication, Xenoarch stone communication, remote talk (to you), recordings, etc?
Does Remote Sight work on silicons? Will you be able to remote see silicons if you hear them talk? What about with a binary chip in your headset?

File: code/datums/mind.dm [hunk 1]
```diff
@@ -69,6 +69,8 @@
 
 	//fix scrying raging mages issue.
 	var/isScrying = 0
+	var/list/heard_before = list()
+	var/list/heard_before_names = list()
```
Comment (ComicIronic) [inline code/datums/mind.dm]: Is this list necessary? You use it in the input(), but it should be possible to pick from the assoc list the same way.

**Comment by 9600bauds** [comment]
I'm not too concerned with who it lets you talk to, barring newcops and such of course, I'm more concerned that it lets you know if they're alive or not when you send the message.

**Comment by ghost** [comment]
@9600bauds And now it doesn't. - I should've added that in the changelog initially, sorry.

File: code/datums/mind.dm (hunk 1 above)
Comment (ghost) [inline code/datums/mind.dm]: Yeah, you were right.

**Comment by ghost** [comment]
@NigglyWiggly 
from the OP:

> You now have to hear someone speak locally - in other words, you have to see their speech bubble, before you can telepathically communicate with them.

I can add exemptions or change that, if you'd like.

**Comment by clusterfack** [comment]
So I cant project my mind into the mime

**Comment by ghost** [comment]
@clusterfack It's a Feature!

**Comment by clusterfack** [comment]
So this works between zlevels too?

**Comment by ghost** [comment]
@clusterfack It should.

**Comment by clusterfack** [comment]
Can you make it so you can for example project mind into people you've examined then

**Comment by ghost** [comment]
you can now examine people to add them to your "heard" list.

**Comment by NigglyWiggly** [comment]
When you say "can" do you mean, like, as an option along with hearing them, or in place of it?

**Comment by ghost** [comment]
@NigglyWiggly An option

**Comment by carshalash** [comment]
Cool, how about we make it so we balance the rest of the race after making their positive somehow even worse than the pda.

**Pull request merged**